lib/Controller/MappingsController.php

```diff
@@ -162,4 +166,127 @@ public function destroy(int $id): JSONResponse
 
         return new JSONResponse([]);
     }
-}
\ No newline at end of file
+
+    /**
+     * Tests a mapping
+     *
+     * This method tests a mapping with provided input data and optional schema validation.
+     *
+     * @NoAdminRequired
+     * @NoCSRFRequired
+     *
+     * @return JSONResponse A JSON response containing the test results
+     *
+     * @example
+     * Request:
+     * {
+     *     "inputObject": "{\"name\":\"John Doe\",\"age\":30,\"email\":\"john@example.com\"}",
+     *     "mapping": {
+	 * 			"mapping": {
+	 * 				"fullName":"{{name}}",
+	 * 				"userAge":"{{age}}",
+	 * 				"contactEmail":"{{email}}"
+	 * 			}
+	 * 	   },
+     *     "schema": "user_schema_id",
+     *     "validation": true
+     * }
+     *
+     * Response:
+     * {
+     *     "resultObject": {
+     *         "fullName": "John Doe",
+     *         "userAge": 30,
+     *         "contactEmail": "john@example.com"
+     *     },
+     *     "isValid": true,
+     *     "validationErrors": []
+     * }
+     */
+    public function test(ObjectService $objectService, IURLGenerator $urlGenerator): JSONResponse
+    {
+		$openRegisters = $objectService->getOpenRegisters();
+
+        // Get all parameters from the request
+        $data = $this->request->getParams();
+
+
+        // Validate that required parameters are present
+        if (!isset($data['inputObject']) || !isset($data['mapping'])) {
+            throw new \InvalidArgumentException('Both `inputObject` and `mapping` are required');
+        }
+
+        // Decode the input object from JSON
+        $inputObject = $data['inputObject'];
+
+        // Decode the mapping from JSON
+		$mapping = $data['mapping'];
+
+        // Initialize schema and validation flags
+        $schema = false;
+        $validation = false;
+
+        // If a schema is provided, retrieve it
+        if (empty($data['schema']) === false) {
+			if ($openRegisters === null) {
+				return new JSONResponse(data: [
+					'error'   => 'Setup error',
+					'message' => 'OpenRegisters must be installed to validate schema.'
+				],statusCode: 412);
+			}
+
+            $schemaId = $data['schema'];
+			try {
+				$schema = $openRegisters->getMapper('schema')->find($schemaId);
+			} catch (DoesNotExistException $exception) {
+				return new JSONResponse(data: [
+					'error' => 'Not found',
+					'message' => 'The specified schema could not be found.',
+				], statusCode: 404);
+			}
+        }
+
+        // Check if validation is requested
+        if (empty($data['validation']) === false) {
+            $validation = $data['validation'];
+        }
+
+        // Create a new Mapping object and hydrate it with the provided mapping
+        $mappingObject = new Mapping();
```

Comment:
Why do we create a new Mapping object?
And if it its purpose is temporary, shouldn't we delete it after testing the mapping?

Reply:
It is never persisted to the database ;)